### PR TITLE
ci: release docker in source only

### PR DESCRIPTION
**Pull request by liyishuai**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
NOROUTE
```

## 新RSS检查列表 / New RSS Script Checklist
  
- [ ] New Route
- [ ] Documentation
  - [ ] CN
  - [ ] EN
- [ ] 全文获取 fulltext
  - [ ] Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] 日期和时间 date and time
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added 
- [ ] `Puppeteer`

### CODE REVIEW

**Comment by TonyRL** [comment]
```diff
jobs:
  release:
+   if: github.repository_owner == 'DIYgod'
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v2
```

**Comment by AgFlore** [comment]
I would rather recommend check whether ${{ secrets.DOCKER_USERNAME }} and ${{ secrets.DOCKER_PASSWORD }} is specified. 

There're users who want to build their own docker images for deployments, and we should not bring unnecessary extra trouble to them.

**Comment by liyishuai** [comment]
> I would rather recommend check whether ${{ secrets.DOCKER_USERNAME }} and ${{ secrets.DOCKER_PASSWORD }} is specified.
> 
> There're users who want to build their own docker images for deployments, and we should not bring unnecessary extra trouble to them.

Good idea! Could you instruct me how to write the conditions? The one I wrote didn't work (liyishuai@c2c718ca6c6f58691bcd275b05591c67e6a14df5):
> The workflow is not valid. .github/workflows/docker-release.yml (Line: 17, Col: 9): Unrecognized named-value: 'secrets'. Located at position 1 within expression: secrets.DOCKER_USERNAME && secrets.DOCKER_PASSWORD

**Comment by AgFlore** [comment]
> > The workflow is not valid. .github/workflows/docker-release.yml (Line: 17, Col: 9): Unrecognized named-value: 'secrets'. Located at position 1 within expression: secrets.DOCKER_USERNAME && secrets.DOCKER_PASSWORD

Ah yes Github Actions does not natively support doing so. One needs to copy the secret to env first, like https://github.com/actions/runner/issues/520#issuecomment-907770967

**Comment by liyishuai** [comment]
Success: https://github.com/liyishuai/RSSHub/actions/runs/1792454129

**Comment by TonyRL** [comment]
Checking the existence of `secrets.DOCKER_USERNAME` seems already enough. For security reasons, I against the idea of putting password/token to envs.

File: .github/workflows/docker-release.yml [hunk 1]
```diff
@@ -12,8 +12,20 @@ on:
...
 jobs:
+  check-env:
+    runs-on: ubuntu-latest
+    outputs:
+      check-docker: ${{ steps.check-docker.outputs.defined }}
+    steps:
+      - id: check-docker
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+        if: "${{ env.DOCKER_USERNAME != '' }}"
```
Comment (TonyRL) [inline .github/workflows/docker-release.yml:23]:
```suggestion
        if: ${{ env.DOCKER_USERNAME != '' }}
```
I think quotes can be skipped

File: .github/workflows/docker-release.yml [hunk 2]
```diff
@@ -12,8 +12,20 @@ on:
...
+        run: echo "::set-output name=defined::true"
   release:
     runs-on: ubuntu-latest
+    needs: [check-env]
```
Comment (TonyRL) [inline .github/workflows/docker-release.yml:27]:
```suggestion
    needs: check-env
```
No need for array.

**Comment by TonyRL** [comment]
Reverted.

`secrets.DOCKER_USERNAME` is set and still skipping build.

See https://github.com/DIYgod/RSSHub/actions/runs/1794535484

File: .github/workflows/docker-release.yml [hunk 3]
```diff
@@ -12,8 +12,20 @@ on:
...
+    steps:
+      - id: check-docker
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+        if: ${{ env.DOCKER_USERNAME != '' }}
+        run: echo "::set-output name=defined::true"
   release:
     runs-on: ubuntu-latest
+    needs: check-env
+    if: needs.check-env.outputs.defined == 'true'
```
Comment (AgFlore) [inline .github/workflows/docker-release.yml:28]: if: needs.check-env.outputs.check-docker == 'true'

Comment (AgFlore) [reply .github/workflows/docker-release.yml:28]: should work like https://github.com/AgFlore/RSSHub/runs/5066207981 @TonyRL @liyishuai

**Pull request merged**